src/main.rs

```diff
@@ -20,29 +51,125 @@ pub struct ParseError {
     pub expected: String,
 }
 
-/// converts and rrule string to a jsonified response
-fn convert_rrule_to_json(
-    rrule_string: String,
-) {
-    let parse_result = RRuleParser::parse(Rule::expr, rrule_string.as_str())
+use pest::iterators::Pair;
+use std::collections::HashMap;
+
+/// converts and rrule string to a rrule struct
+fn convert_to_rrule<'a>(rrule_result: &mut RRule<'a>, rrule_string: &'a str) {
+    let parse_result = RRuleParser::parse(Rule::expr, rrule_string)
         .expect("unsuccessful parse") // unwrap the parse result
-        .next().unwrap();
+        .next()
+        .unwrap();
```

Comment:
`.map_err` is your friend with https://boats.gitlab.io/failure/ to handle parsing errors.